libraries/botbuilder-ai/src/qnaMakerRecognizer.ts

```diff
@@ -292,4 +294,81 @@ export class QnAMakerRecognizer extends Recognizer implements QnAMakerRecognizer
         const logPersonalInfo = this.logPersonalInformation.getValue(dc.state);
         return new QnAMaker(endpoint, {}, this.telemetryClient, logPersonalInfo);
     }
+
+    /**
+     * Gets an instance of [QnAMakerClient](xref:botbuilder-ai.QnAMakerClient)
+     *
+     * @param {DialogContext} dc The dialog context used to access state.
+     * @returns {QnAMakerClient} An instance of QnAMakerClient.
+     */
+    protected getQnAMakerClient(dc: DialogContext): QnAMakerClient {
+        const qnaClient = dc.context?.turnState?.get(QnAMakerClientKey);
+        if (qnaClient) {
+            return qnaClient;
+        }
+
+        const { value: endpointKey, error } = this.endpointKey?.tryGetValue(dc.state) ?? {};
+        if (!endpointKey || error) {
+            throw new Error(`Unable to get a value for endpointKey from state. ${error}`);
+        }
+        const { value: host, error: error2 } = this.hostname?.tryGetValue(dc.state) ?? {};
+        if (!host || error2) {
+            throw new Error(`Unable to get a value for hostname from state. ${error2}`);
+        }
+        const { value: knowledgeBaseId, error: error3 } = this.knowledgeBaseId?.tryGetValue(dc.state) ?? {};
+        if (!knowledgeBaseId || error3) {
+            throw new Error(`Unable to get a value for knowledgeBaseId from state. ${error3}`);
+        }
+
+        const endpoint: QnAMakerEndpoint = { endpointKey, host, knowledgeBaseId };
+        const logPersonalInfo = this.logPersonalInformation.getValue(dc.state);
+        return new QnAMaker(endpoint, {}, this.telemetryClient, logPersonalInfo);
+    }
+
+    /**
+     * Uses the RecognizerResult to create a list of properties to be included when tracking the result in telemetry.
+     *
+     * @param {RecognizerResult} recognizerResult Recognizer result.
+     * @param {Record<string, string>} telemetryProperties A list of properties to append or override the properties created using the RecognizerResult.
+     * @param {DialogContext} dialogContext Dialog context.
+     * @returns {Record<string, string>} A dictionary that can be included when calling the trackEvent method on the TelemetryClient.
+     */
+    protected fillRecognizerResultTelemetryProperties(
+        recognizerResult: RecognizerResult,
+        telemetryProperties: Record<string, string>,
+        dialogContext?: DialogContext
```

Comment:
Can we make this a required parameter since there's an error thrown below if it isn't provided?